openquake/hazardlib/tests/geo/surface/gridded_test.py

```diff
@@ -70,10 +71,17 @@ def test_get_ry0_distance(self):
                           self.mesh)
 
     def test_get_strike(self):
-        np.testing.assert_equal(np.nan, self.surf.get_strike())
+        strike = self.surf.get_strike()
+        strike_2 = self.surf2.get_strike()
+        strike_3 = self.surf3.get_strike()
+        self.assertEqual(strike, 0.707115755836035)
+        self.assertEqual(strike_2, 0.7070046277015921)
+        self.assertEqual(strike_3, 179.49997899960738)
+
 
-    def test_get_dip(self):
-        np.testing.assert_equal(np.nan, self.surf.get_dip())
+    def test_dip(self):   
+        dip = self.surf.get_dip()
+        self.assertEqual(dip, 90.4742185763942)
```

Comment:
The dip angle is [0, 90] how is this possible?

Reply:
yes it was showing greater than 90 because the depth or z component of 'n' was negative. I have rectified it.